presto/utils.py

```diff
@@ -216,88 +180,111 @@ def process_parquet(df: pd.DataFrame) -> pd.DataFrame:
         "OPTICAL-B8A",
     ]
     bands100m = ["METEO-precipitation_flux", "METEO-temperature_mean"]
-
-    df["timestamp_ind"] = (df["timestamp"].dt.year * 12 + df["timestamp"].dt.month) - (
-        df["start_date"].dt.year * 12 + df["start_date"].dt.month
-    )
-    df["valid_position"] = (df["valid_date"].dt.year * 12 + df["valid_date"].dt.month) - (
-        df["start_date"].dt.year * 12 + df["start_date"].dt.month
-    )
-    df["valid_position_diff"] = df["timestamp_ind"] - df["valid_position"]
-
-    # save the initial start_date for later
-    df["initial_start_date"] = df["start_date"].copy()
-    index_columns.append("initial_start_date")
-
-    # define samples where valid_date is outside the range of the actual extractions
-    # and remove them from the dataset
-    latest_obs_position = df.groupby(["sample_id"])[
-        ["valid_position", "timestamp_ind", "valid_position_diff"]
-    ].max()
-    df["is_last_available_ts"] = (
-        df["sample_id"].map(latest_obs_position["timestamp_ind"]) == df["timestamp_ind"]
-    )
-    samples_after_end_date = latest_obs_position[
-        (latest_obs_position["valid_position"] > latest_obs_position["timestamp_ind"])
-    ].index
-    samples_before_start_date = latest_obs_position[
-        (latest_obs_position["valid_position"] < 0)
-    ].index
-
-    if len(samples_after_end_date) > 0 or len(samples_before_start_date) > 0:
-        logger.warning(
-            f"""\
-Dataset {df["ref_id"].iloc[0]}: removing {len(samples_after_end_date)} \
-samples with valid_date after the end_date \
-and {len(samples_before_start_date)} samples with valid_date before the start_date"""
+    static_features = ["DEM-alt-20m", "DEM-slo-20m", "lat", "lon"]
+
+    feature_columns = bands10m + bands20m + bands100m
+    # for index columns we need to include all columns that are not feature columns
+    index_columns = [col for col in df.columns if col not in feature_columns]
+    # and also ensure that static DEM columns and lat-lon are included.
+    # if they are not available in the DataFrame,
+    # they will be initialized with NODATAVALUE
+    index_columns.extend(static_features)
+    index_columns.remove("timestamp")
+
+    # check that all feature columns are present in the DataFrame
+    # or initialize them with NODATAVALUE
+    for feature_col in feature_columns + static_features:
+        if feature_col not in df.columns:
+            df[feature_col] = NODATAVALUE
+
+    df["timestamp_ind"] = df.groupby("sample_id")["timestamp"].rank().astype(int)
+
+    # Assign start_date and end_date as the minimum and maximum available timestamp
+    df["start_date"] = df["sample_id"].map(df.groupby(["sample_id"])["timestamp"].min())
+    df["end_date"] = df["sample_id"].map(df.groupby(["sample_id"])["timestamp"].max())
+    index_columns.extend(["start_date", "end_date"])
+
+    if use_valid_time:
+        if "valid_date" in df.columns:
```

Comment:
I think it's simpler to put somewhere higher up the renaming, regardless of whether the old column name is there or not:
`df.rename(columns={"valid_date": "valid_time"}, inplace=True)`
I tested it and when it's not there it does nothing but also doesn't raise an exception.

So you can put the comment on old naming and the renaming higher up, then here we don't need the `if`, and we don't need to remove any element from the `index_columns` if we've done the potential renaming before constructing this list.

Reply:
done 5d85e5a9429b3590fc49319033ab65c3c1489701